pkg/controller/daemon/daemon_controller.go

```diff
@@ -1327,11 +1342,48 @@ func NewPod(ds *extensions.DaemonSet, nodeName string) *v1.Pod {
 	return newPod
 }
 
+// nodeSelectPredicates are the predicates that used to select node candidates for DaemonSet
+func nodeSelectPredicates(pod *v1.Pod, meta algorithm.PredicateMetadata, nodeInfo *schedulercache.NodeInfo) (bool, []algorithm.PredicateFailureReason, error) {
+	var predicateFails []algorithm.PredicateFailureReason
+	fit, reasons, err := predicates.PodFitsHost(pod, meta, nodeInfo)
+	if err != nil {
+		return false, predicateFails, err
+	}
+	if !fit {
+		predicateFails = append(predicateFails, reasons...)
+	}
+
+	fit, reasons, err = predicates.PodMatchNodeSelector(pod, meta, nodeInfo)
+	if err != nil {
+		return false, predicateFails, err
+	}
+	if !fit {
+		predicateFails = append(predicateFails, reasons...)
+	}
+	return len(predicateFails) == 0, predicateFails, nil
+}
+
 // Predicates checks if a DaemonSet's pod can be scheduled on a node using GeneralPredicates
 // and PodToleratesNodeTaints predicate
 func Predicates(pod *v1.Pod, nodeInfo *schedulercache.NodeInfo) (bool, []algorithm.PredicateFailureReason, error) {
 	var predicateFails []algorithm.PredicateFailureReason
-	critical := utilfeature.DefaultFeatureGate.Enabled(features.ExperimentalCriticalPodAnnotation) && kubelettypes.IsCriticalPod(pod)
+
+	// If NoDaemonSetScheduler is enable, only check nodeSelector and nodeAffinity.
+	if utilfeature.DefaultFeatureGate.Enabled(features.NoDaemonSetScheduler) {
+		// EssentialPredicates includes PodFitsHost, PodFitsHostPorts and PodMatchNodeSelector
```

Comment:
We don't need to run PodFitsHostPorts and actually your function does not run it. Please remove it from the comment.